lib/Service/SttService.php

```diff
@@ -22,56 +22,178 @@
-		return $this->manager->transcribeFile($audioFile, $userId, Application::APP_ID);
+
+		if ($schedule) {
+			$this->manager->scheduleFileTranscription($audioFile, $userId, Application::APP_ID);
+			return 'ok';
+		}
+
+		return $this->manager->transcribeFile($audioFile);
 	}
 
 	/**
-	 * @param string $audioContent
+	 * @param string $tempFileLocation
+	 * @param bool $schedule
 	 * @param string|null $userId
 	 * @return string The transcript
 	 * @throws NotPermittedException
 	 * @throws PreConditionNotMetException
 	 * @throws InvalidArgumentException
+	 * @throws RuntimeException
 	 */
-	public function transcribeAudio(string $audioContent, ?string $userId): string {
-		$audioFile = $this->getFileObject($userId ?? '', $audioContent);
-		return $this->manager->transcribeFile($audioFile, $userId, Application::APP_ID);
+	public function transcribeAudio(string $tempFileLocation, bool $schedule, ?string $userId): string {
+		if ($userId === null) {
+			throw new InvalidArgumentException('userId must not be null');
+		}
+
+		$audioFile = $this->getFileObject($userId, $tempFileLocation);
+
+		if ($schedule) {
+			$this->manager->scheduleFileTranscription($audioFile, $userId, Application::APP_ID);
+			return 'ok';
+		}
+
+		return $this->manager->transcribeFile($audioFile);
+	}
+
+	/**
+	 * @param string $userId
+	 * @param string $tempFileLocation
+	 * @return File
+	 * @throws NotPermittedException
+	 * @throws RuntimeException
+	 */
+	private function getFileObject(string $userId, string $tempFileLocation): File {
+		$userFolder = $this->rootFolder->getUserFolder($userId);
+
+		$sttFolderName = $this->config->getAppValue(Application::APP_ID, 'stt_folder', '(not set)');
+		if ($sttFolderName === '(not set)') {
+			$sttFolderName = Application::REC_FOLDER;
+
+			if ($userFolder->nodeExists($sttFolderName)) {
+				$sttFolder = $this->getRandomNamedFolder($userId);
+				$sttFolderName = $sttFolder->getName();
+			} else {
+				$sttFolder = $userFolder->newFolder($sttFolderName);
+			}
+			$this->config->setAppValue(Application::APP_ID, 'stt_folder', $sttFolderName);
+		} else {
+			$sttFolder = $userFolder->get($sttFolderName);
+			if (!$sttFolder instanceof Folder) {
+				// the folder created by this app was tampered with
+				// create a new one
+				$sttFolder = $this->getRandomNamedFolder($userId);
+				$sttFolderName = $sttFolder->getName();
+				$this->config->setAppValue(Application::APP_ID, 'stt_folder', $sttFolderName);
+			}
+		}
+
+		$filename = (new DateTime())->format('d-M-Y-Hisu') . '.mp3';
+
+		$audioData = file_get_contents($tempFileLocation);
+		$audioFile = $sttFolder->newFile($filename, $audioData);
+
+		return $audioFile;
+	}
+
+	/**
+	 * @param string $userId
+	 * @param integer $try
+	 * @return Folder
+	 * @throws RuntimeException
+	 */
+	private function getRandomNamedFolder(string $userId, int $try = 3): Folder {
+		$userFolder = $this->rootFolder->getUserFolder($userId);
+		$randomId = bin2hex(random_bytes(4));
```

Comment:
mmmh, random characters are a bit confusing for something user-facing I'd say. Maybe just increment a number?